pkg/operator/resource_reconciler.go

```diff
@@ -374,6 +401,36 @@ func (rr *ResourceReconciler) onStatefulSetUpdate(old, cur *appsv1.StatefulSet)
 	rr.EnqueueForReconciliation(obj)
 }
 
+func (rr *ResourceReconciler) onDaemonSetUpdate(old, cur *appsv1.DaemonSet) {
+	level.Debug(rr.logger).Log("msg", "update handler", "resource", "daemonset", "old", old.ResourceVersion, "cur", cur.ResourceVersion)
+
+	if rr.DeletionInProgress(cur) {
+		return
+	}
+
+	if !rr.hasObjectChanged(old, cur) {
+		return
+	}
+
+	obj := rr.syncer.Resolve(cur)
+	if obj == nil {
+		return
+	}
+
+	level.Debug(rr.logger).Log("msg", "DaemonSet updated")
+	rr.metrics.TriggerByCounter("DaemonSet", UpdateEvent).Inc()
+
+	if !rr.hasStateChanged(old, cur) {
+		// If the statefulset state (spec, labels or annotations) hasn't
+		// changed, the operator can only update the status subresource instead
+		// of doing a full reconciliation.
+		rr.EnqueueForStatus(obj)
```

Comment:
@simonpasquier Now we haven't supported status subresource for DaemonSet yet, so we cannot use it for now right?

Reply:
Ok I think I got it now. We haven't supported status subresource for Prometheus Agent in DaemonSet mode. But DaemonSet's status has already been supported by K8s. So we can `EnqueueForStatus()` for DaemonSet, right?

Reply:
AFAICT we don't support the status subresource when mode=daemonset so it's not worth calling EnqueueForStatus() but keep the if loop and add a TODO comment that we can call EnqueueForStatus() when implementing the status.